### ISSUE TITLE
Terminal connection: Denied!

### ISSUE THREAD

**Issue by D4V3M0NK** [Ylianst/MeshCentral#4057]
Running MC2 v0.9.98
A new one on me as of this morning ... a remote Linux system that I was attempting to log into via the terminal, showed me a "Denied" message. All my remote systems are headless and have been set up without any requirement from a remote user to connect, so this took me for a bit of a loop.

![MC2 Terminal Denied screenshot](https://user-images.githubusercontent.com/7572742/171258364-68a24fcb-1f91-4c2d-b00b-5aaa1126d9f1.png)

Attempting to connect via Files to this remote works without issue, and all other remotes permit me to log in via Terminal as expected. Also, when using the SSH agent (via MC Router) to connect to this one specific remote, works without issue. When I inspect the Events on this remote, it simply shows me a "started terminal session" then "ended terminal session" immediately afterwards.

Thoughts?

**Comment by D4V3M0NK** [comment]
_(Note to self : tested against VHAV20PORCCAVLTT)_

1. "undefined" 
![Screenshot from 2022-06-21 14-43-38](https://user-images.githubusercontent.com/7572742/174902387-3d283c15-12b4-49b2-aafe-e75ec1ad1186.png)

3. The agent shows "Setup..." and that's it
![Screenshot from 2022-06-21 14-44-24](https://user-images.githubusercontent.com/7572742/174902444-fac7e36a-805e-4fd4-9a6d-64bb24aab3c0.png)
I've never used the Console > Agent Actions before, so not sure if there was supposed to be any response when I Cleared the core and uploaded the recovery core, but I didn't get any response when I selected those options

**Comment by krayon007** [comment]
Interesting... You can click on clear core, and select upload default core, to restore the default. This time for terminal, can you right click on the connect button and select login shell and tell me if that works?

It almost seems as if /bin/sh and/or /bin/bash was missing. Login shell will spawn the login shell instead.

**Comment by D4V3M0NK** [comment]
@krayon007 - there's no option for right clicking on the Connect button (gives me the default browser options).

![MC2 Terminal Right Click](https://user-images.githubusercontent.com/7572742/174922461-1eb5330c-d66c-4537-9719-67b178d0c56c.png)

You mention login config... on this server, I do have the following configured if it's applicable:
```
...
"terminal": {
          "linuxshell": "login",
          "default": "login"
      },
...
```

**Comment by silversword411** [comment]
> ![MC2 Terminal Right Click](https://user-images.githubusercontent.com/7572742/174922461-1eb5330c-d66c-4537-9719-67b178d0c56c.png)

There's something off with your browser/right-click then

![2022-06-21_212103 - meshrightclick](https://user-images.githubusercontent.com/13395348/174923225-a904faef-a474-4e15-a453-8e16976af001.png)

**Comment by krayon007** [comment]
@D4V3M0NK which browser are you using?

**Comment by D4V3M0NK** [comment]
ChromeOn Jun 21, 2022 9:08 PM, Bryan Roe ***@***.***> wrote:
@D4V3M0NK which browser are you using?

**Comment by D4V3M0NK** [comment]
@krayon007 - this morning I tried selecting the same remote system and then "Reset device" (is this a reboot?) ... tried the terminal again, and I get my login prompt. Tried on another remote where I'm having the same issue, and this also worked.

1. Thoughts? Please bear in mind that my remotes only have access to the MC2 server, they have no other internet access.
2. The browser worries me a little. Are these controls new additions to MC2 (I'm running 0.9.98)?

**Comment by D4V3M0NK** [comment]
Hmmm ... I've tried on both Windows and Ubuntu, Chrome and Edge, right clicking still only gives me the standard browser options...

**Comment by silversword411** [comment]
> Hmmm ... I've tried on both Windows and Ubuntu, Chrome and Edge, right clicking still only gives me the standard browser options...


Do you do something special with your environments? Like plugins, have you tested with an empty or system not configured by you?

**Comment by D4V3M0NK** [comment]
@krayon007 : I had my customer reboot one of the systems that's coming up Denied and that solved my problem. However Bryan, can you confirm that "Reset device" is the same as a reboot? I'm hoping to at least reboot all my other systems that are showing "Denied" without customer intervention.

**Comment by krayon007** [comment]
I'll double check after lunch, but I'm pretty sure reset device was just a power action for reboot.

**Comment by D4V3M0NK** [comment]
Also Bryan, I have the following set up in my `config.json`:
```
"terminal": {
          "linuxshell": "login",
          "default": "login"
      },
```
I've just disabled that and sure enough, I see the same browser control that you do!

**Comment by krayon007** [comment]
When you see that control, do any of the options work, or do they all say denied?

**Comment by D4V3M0NK** [comment]
If I upgrade the server, the agents self-upgrade, correct?

**Comment by krayon007** [comment]
Yes. Unless the agents are configured to not update. But the default is to allow auto updates.
